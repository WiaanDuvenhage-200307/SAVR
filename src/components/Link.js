// import React, {useState} from 'react';

// const Link = ({page, children}) => {

//     const [status, setStatus] = useState("normal");

//     // (condition) ? do this if true : do this if false

//     return (
//         <a href={page} onMouseEnter={() => setStatus("hovered")} onMouseLeave={() => setStatus("normal")}>
//             {children}
//         </a>
//     );
// };

// export default Link;