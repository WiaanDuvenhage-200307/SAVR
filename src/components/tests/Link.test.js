import renderer from 'react-test-renderer';
import Link from '../classActivities/Link';

// Test to see if our component rendered correctly
describe('Testing the link component....', () => { 

    test("change the color of the anchor tag...", () => {
        //1. Create/render our component (as you would in the return of a component)
        const component = renderer.create(
            <Link page="https://facebook.com">Facebook</Link>
        );

        // Rendering the component as json so we can investigate it
        let tree = component.toJSON();

        // Test to see if our component renders correctly - check did it render correctly by just creating the
        expect(tree).toMatchSnapshot();

        // 2. We simulate interaction with the component

        // trigger the hover of our component
        renderer.act(() => {
            tree.props.onMouseEnter();
        });

        // re-render the tree
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test("change the color of the anchor tag on leave...", () => {
        //1. Create/render our component (as you would in the return of a component)
        const component = renderer.create(
            <Link page="https://facebook.com">Facebook</Link>
        );

        // Rendering the component as json so we can investigate it
        let tree = component.toJSON();

        // Test to see if our component renders correctly - check did it render correctly by just creating the
        expect(tree).toMatchSnapshot();

        // 2. We simulate interaction with the component

        // trigger the hover of our component
        renderer.act(() => {
            tree.props.onMouseLeave();
        });

        // re-render the tree
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

})
