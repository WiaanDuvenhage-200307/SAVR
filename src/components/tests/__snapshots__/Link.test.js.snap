// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the link component.... change the color of the anchor tag on leave... 1`] = `
<a
  href="https://facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Testing the link component.... change the color of the anchor tag on leave... 2`] = `
<a
  href="https://facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Testing the link component.... change the color of the anchor tag... 1`] = `
<a
  href="https://facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Testing the link component.... change the color of the anchor tag... 2`] = `
<a
  href="https://facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
